woocommerce/woocommerce-ios
# Update Products tab Filter CTA title based on the number of active filters

Products tab Filter CTA title for #2037

I realized I pushed the branch yesterday but forgot to make a PR 😅 

## Changes

- Added `numberOfActiveFilters: Int` to `FilterListViewModel` as a replacement of `numberOfActiveFilters` via the view model's `filterTypeViewModels` + unit tests for `FilterProductListViewModel`
- In `ProductsViewController` (products tab), kept a reference to the filter CTA `filterButton`, and updated its title on product change

## Testing

- Go to the Products tab
- Tap on the "Filter" CTA in the top toolbar
- Select a filter type and an option
- Tap "Show Products" at the bottom --> it should go back to the Products tab with "Filter (1)" in the toolbar where it used to say "Filter", with a list of products matching the selected filter
- Tap on the "Filter (1)" CTA
- Select another filter type and option
- Tap "<" in the navigation bar
- Tap "Show Products" at the bottom --> it should go back to the Products tab with "Filter (2)" in the toolbar where it used to say "Filter (1)", with a list of products matching the selected filters
- Tap on the "Filter (2)" CTA
- Tap "Clear all" in the navigation bar --> the filters should be cleared
- Tap "Show Products" at the bottom --> it should go back to the Products tab with "Filter" in the toolbar where it used to say "Filter (2)", with all the products listed

## Example screencast/screenshots

![simulator](https://user-images.githubusercontent.com/1945542/81135465-e47ede80-8f8a-11ea-92c9-20ee8aeeeb08.gif)


no filters | 1 filter | 2 filters | 3 filters + no results
-- | -- | -- | --
![Simulator Screen Shot - iPhone 11 - 2020-05-06 at 11 15 33](https://user-images.githubusercontent.com/1945542/81135513-1bed8b00-8f8b-11ea-95a1-8d9663360832.png) | ![Simulator Screen Shot - iPhone 11 - 2020-05-06 at 11 15 44](https://user-images.githubusercontent.com/1945542/81135519-1f811200-8f8b-11ea-8d54-955f08abb5e1.png) | ![Simulator Screen Shot - iPhone 11 - 2020-05-06 at 11 15 50](https://user-images.githubusercontent.com/1945542/81135524-20b23f00-8f8b-11ea-8407-735e8dca3571.png) | ![Simulator Screen Shot - iPhone 11 - 2020-05-06 at 11 16 05](https://user-images.githubusercontent.com/1945542/81135525-214ad580-8f8b-11ea-85c4-a30e7437134e.png)


Update release notes:
